### ISSUE TITLE
GenerateConfiguration() Go binding does not accept YAML based config

### ISSUE THREAD

**Issue by nmagnezi** [nmstate/nmstate#1970]
## Describe the bug
The [GenerateConfiguration()](https://github.com/nmstate/nmstate/blob/8f3a58b3069a92ec4ef43b548dc5f73c62f6c315/rust/src/go/nmstate/nmstate.go#L211) Go binding does not accept `YAML` formatted configurations.
I'm trying to use it instead of invoking `nmstatectl gc` via code (currently using a `YAML` string), which accepts both `YAML` and `JSON` according to [nmstate documentation](https://nmstate.io/features/gen_conf.html).

## Nmstate version
- nmstatectl 2.1.2
- imported code from master

## Platform
- Platform: CentOS 9 Stream

## How reproducible
- 100%

## Steps to Reproduce
1. Used this code to reproduce the issue: https://gist.github.com/nmagnezi/00fa21f72e0046efd06c732320b76cc6
2. Took the example from nmstate unit-tests: and converted to `YAML` https://github.com/nmstate/nmstate/blob/8f3a58b3069a92ec4ef43b548dc5f73c62f6c315/rust/src/go/nmstate/nmstate_test.go#L47-L61

## Nmstate output
- Expected output:

- Current output:
```
failed when generating the configuration ---
interfaces:
- name: dummy1
  state: up
  type: dummy
 with rc: 1, err_msg: Invalid json string: invalid number at line 1 column 2, err_kind: InvalidArgument
```

**Comment by cathay4t** [comment]
The go binding is using JSON as API because we are considering yaml not fit for API due to [yaml 1.1 vs 1.2 issue][1].

@qinqon Can we somehow mention go binding is JSON API somwhere?

[1]: https://github.com/yaml/pyyaml/issues/98

**Comment by cathay4t** [comment]
Meanwhile, the `serde_yaml` we used for CLI in nmstate is not stable enough comparing to `serde_json`.

**Comment by qinqon** [comment]
> The go binding is using JSON as API because we are considering yaml not fit for API due to [yaml 1.1 vs 1.2 issue](https://github.com/yaml/pyyaml/issues/98).
> 
> @qinqon Can we somehow mention go binding is JSON API somwhere?

Maybe we can have similar approach to CLI and pass some argument to clib to use json or yaml ? we can even discover if it's json or yaml by checking if there is "{" at the beggining.

**Comment by ffmancera** [comment]
I agree with the solution proposed by @qinqon.. IMO, we should support YAML.. the same way we support it on python binding o rust
